README.md

```diff
@@ -15,23 +15,43 @@ npm install breadth-filter
 ```js
 const breadthFilter = require('breadth-filter')
 
-const filtered = breadthFilter({
+const data = {
   user: {
     name: 'someone',
     password: 'THIS SHOULD BE SECRET!!'
   }
-}, (value, key, path) => {
+}
+
+function onValue (value, key, path) {
   if (key === 'password') {
     console.log('redacted field at', path.join('.'))
     return '[redacted]'
   }
   return value
+}
+
+// build a new filtered object...
+
+const filtered = breadthFilter(data, {
+  onValue
+})
+
+// or, mutate in-place and break cycles...
+
+breadthFilter(data, {
+  onValue,
+  onObject (value, key, path, isNew) {
```

Comment:
Not sure how much you want to do with the docs for this, but if you want to make this module easy to use you should probably document the arguments - e.g `key` can be `null` etc

Reply:
fixed